django/thunderstore/modpacks/models/legacyprofile.py

```diff
@@ -0,0 +1,51 @@
+import ulid2
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.core.files.storage import get_storage_class
+from django.core.files.uploadedfile import TemporaryUploadedFile
+from django.db import models
+from django.db.models import Sum
+
+from thunderstore.core.mixins import TimestampMixin
+
+LEGACYPROFILE_STORAGE_CAP = (
+    1024 * 1024 * 1024 * settings.LEGACYPROFILE_MAX_TOTAL_SIZE_GB
+)
+
+
+def get_legacy_profile_file_path(instance: "LegacyProfile", filename: str):
+    return f"modpacks/legacyprofile/{instance.id}"
+
+
+class LegacyProfileManager(models.Manager):
+    def create_from_upload(self, content: TemporaryUploadedFile) -> "LegacyProfile":
+        if content.size + self.get_total_used_disk_space() > LEGACYPROFILE_STORAGE_CAP:
+            raise ValidationError(
+                f"The server has reached maximum total storage used, and can't receive new uploads"
+            )
+
+        return self.create(
+            file=content,
+            file_size=content.size,
+        )
+
+    def get_total_used_disk_space(self) -> int:
+        return self.aggregate(total=Sum("file_size"))["total"] or 0
+
+
+class LegacyProfile(TimestampMixin, models.Model):
+    objects: "LegacyProfileManager[LegacyProfile]" = LegacyProfileManager()
+
+    id = models.UUIDField(
+        default=ulid2.generate_ulid_as_uuid,
+        primary_key=True,
+        editable=False,
+    )
+    file = models.FileField(
+        upload_to=get_legacy_profile_file_path,
+        storage=get_storage_class(settings.PACKAGE_FILE_STORAGE)(),
```

Comment:
Should this use the newly created `MODPACK_FILE_STORAGE` instead of `PACKAGE_FILE_STORAGE`?

Reply:
It should yes, fixed